pkg/ottl/ottlfuncs/func_sort.go

```diff
@@ -0,0 +1,219 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package ottlfuncs // import "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl/ottlfuncs"
+
+import (
+	"cmp"
+	"context"
+	"fmt"
+	"slices"
+	"strconv"
+
+	"go.opentelemetry.io/collector/pdata/pcommon"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl"
+)
+
+const (
+	sortAsc  = "asc"
+	sortDesc = "desc"
+)
+
+type SortArguments[K any] struct {
+	Target ottl.Getter[K]
+	Order  ottl.Optional[string]
+}
+
+func NewSortFactory[K any]() ottl.Factory[K] {
+	return ottl.NewFactory("Sort", &SortArguments[K]{}, createSortFunction[K])
+}
+
+func createSortFunction[K any](_ ottl.FunctionContext, oArgs ottl.Arguments) (ottl.ExprFunc[K], error) {
+	args, ok := oArgs.(*SortArguments[K])
+
+	if !ok {
+		return nil, fmt.Errorf("SortFactory args must be of type *SortArguments[K]")
+	}
+
+	order := sortAsc
+	if !args.Order.IsEmpty() {
+		o := args.Order.Get()
+		switch o {
+		case sortAsc, sortDesc:
+			order = o
+		default:
+			return nil, fmt.Errorf("invalid arguments: %s. Order should be either \"%s\" or \"%s\"", o, sortAsc, sortDesc)
+		}
+	}
+
+	return Sort(args.Target, order)
+}
+
+func Sort[K any](target ottl.Getter[K], order string) (ottl.ExprFunc[K], error) {
```

Comment:
```suggestion
func sort[K any](target ottl.Getter[K], order string) (ottl.ExprFunc[K], error) {
```
We should unexport this since `NewSortFactory` is the only necessary public API for this function.